x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java

```diff
@@ -70,6 +70,24 @@
                         new RoleDescriptor.IndicesPrivileges[] {
                                 RoleDescriptor.IndicesPrivileges.builder().indices(".monitoring-*").privileges("all").build() },
                         null, MetadataUtils.DEFAULT_RESERVED_METADATA))
+                .put("stack_monitoring_agent", new RoleDescriptor(
+                        "stack_monitoring_agent",
+                        new String[] {
+                            "monitor",
+                            "manage_index_templates"
```

Comment:
@ycombinator can you please explain why `manage_index_templates` is required ?
Isn't `.indices(".monitoring-*", "metricbeat-*").privileges("index", "create_index")` enough?

Reply:
> can you please explain why manage_index_templates is required ?

When Metricbeat starts up today, it checks if the Metricbeat index template exists and, if not, tries to create it.

> Isn't .indices(".monitoring-*", "metricbeat-*").privileges("index", "create_index") enough?

Creating templates is a cluster-level privilege so having these index-level privileges is not enough.

Reply:
One thing I could do here to tighten things up a bit is to specifically grant the `indices:admin/template/get` and `indices:admin/template/put` cluster-level privileges instead of the broader `manage_index_templates` cluster-level privilege.

Reply:
What I was thinking about, was, do you need both to `put template` and `create index`. Cannot Metricbeat use one or the other. Preferably `create index` because we have tighter authz for it (the privilege has the index name pattern), although in this case, when for example there is a centralized monitoring cluster, a template would be more suitable.

> [EDIT] I've done this in a33449a. Let me know what you think.

So, you've removed the `delete` action but `put` is an overwrite (our fault not yours 😃). It's fine either way.